auraed/src/runtime/cell_service/executables/executable.rs

```diff
@@ -75,6 +126,117 @@ impl Executable {
 
         Some(process.pid())
     }
+
+    /// Returns any unread lines from stdout if [Executable] is running, otherwise returns an empty
+    /// [Vec].
+    pub fn read_stdout(&mut self) -> io::Result<Vec<String>> {
```

Comment:
I think these should be on `Process`. Access to the inner `procfs::process::Process` or `Child` is a result of `Process` being an enum and enum variants always(?) being pub.

As of now, `Executable` always results a `Process::Spawned` and `NestedAuraed` in a `Process::Cloned`. Adding to `Process` allows us to share logic between both (e.g., `kill`).

We should probably make `Process` in a struct with an inner emum (`ProcessInner` or something) to guide the correct usage.

Reply:
moving them.. this might also help solve my thread issue as the process can be Arc i think and we can access it from a thread running on Executable's scope.